comps/retrievers/src/integrations/redis.py

```diff
@@ -101,6 +101,9 @@ async def invoke(
         if logflag:
             logger.info(input)
 
+        if input.index_name:
+            self.client = self._initialize_client(index_name=input.index_name)
```

Comment:
What is the intended behavior of specifying an index name? Right now it seems like if you specify an index name that's going to stick forever until you specify a different name (vs having that index name just affect your one search and then going back to the default index name).